examples/SPDXJSONExample-v2.0.json

```diff
@@ -2,380 +2,267 @@
+      "copyrightText" : "Copyright 2008-2010 John Smith",
+      "fileContributor" : [ "The Regents of the University of California", "Modified by Paul Mundt lethal@linux-sh.org", "IBM Corporation" ],
+      "fileName" : "./package/foo.c",
+      "fileType" : [ "SOURCE" ],
+      "licenseComments" : "The concluded license was taken from the package level that the file was included in.",
+      "licenseConcluded" : "(LGPL-2.0-only OR LicenseRef-0)",
+      "licenseInfoInFile" : [ "GPL-2.0-only", "LicenseRef-0" ],
+      "noticeText" : "Copyright (c) 2001 Aaron Lehmann aaroni@vitelus.com\n\nPermission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the �Software�), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions: \nThe above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.\n\nTHE SOFTWARE IS PROVIDED �AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE."
+    } ],
+    "snippets" : [ {
+      "comment" : "This snippet was identified as significant and highlighted in this Apache-2.0 file, when a commercial scanner identified it as being derived from file foo.c in package xyz which is licensed under GPL-2.0.",
+      "copyrightText" : "Copyright 2008-2010 John Smith",
+      "licenseComments" : "The concluded license was taken from package xyz, from which the snippet was copied into the current file. The concluded license information was found in the COPYING.txt file in package xyz.",
+      "licenseConcluded" : "GPL-2.0",
+      "licenseInfoInSnippet" : [ "GPL-2.0-only" ],
+      "name" : "from linux kernel",
+      "range" : [ {
+        "endPointer" : {
+          "lineNumber" : "23",
+          "reference" : "SPDXRef-DoapSource"
+        },
+        "startPointer" : {
+          "lineNumber" : "5",
+          "reference" : "SPDXRef-DoapSource"
+        }
+      }, {
+        "endPointer" : {
+          "offset" : "420",
+          "reference" : "SPDXRef-DoapSource"
+        },
+        "startPointer" : {
+          "offset" : "310",
+          "reference" : "SPDXRef-DoapSource"
+        }
+      } ],
+      "snippetFromFile" : "SPDXRef-DoapSource"
```

Comment:
I'm guessing this will be one of those "wait until 3.0" sort of questions  :)  but I'll ask anyway:

I find it kind of confusing that for Files contained by Packages, there's a `hasFile` property at the Package level -- but for Snippets, there's a `snippetFromFile` property at the Snippet level instead. Should this be flipped so that it's instead `hasSnippets` at the File level, for consistency?

Reply:
For the use cases I deal with, when creating a Snippet for an SPDX document, you know what file it belongs to.  When creating a File, you don't know in advance what snippets belong to it so the relationship needs to be stored in the snippet and not in the file.  You may not even have a File to modify to include the Snippet reference if the File is external to the SPDX document.

Reply:
That's a valid point, I hadn't considered that. Thanks @goneall!